### PR TITLE
Add Python 2.7 support

### PR DESCRIPTION
**Pull request by gordonje**
Adding support for Python 2 with minimum impact. This entails:

* Adding an explicit namespace for `opencivicdata` (to continue combining this package and python-opencivicdata-django under one namespace).
* Adding Python 2 unicode compatibility to csv reading (using `backports.csv`).
* Enabling tests for Python 2.7.

I also added a `requirements-test.txt` because I was hitting a version conflict with py.test in Travis (example [here](https://travis-ci.org/california-civic-data-coalition/python-opencivicdata-divisions/jobs/209112441)). Not totally sure why this fix works, but it has the advantages of being consist with how python-opencivicdata-django is set up.

This change is required in order to add Python 2 support for python-opencivicdata-django (see opencivicdata/python-opencivicdata-django/76).

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -11,7 +11,9 @@
       long_description="",
       url="",
       packages=['opencivicdata.divisions'],
+      namespace_packages=['opencivicdata'],
       include_package_data=True,
+      install_requires=['backports.csv'],
```
Comment (jamesturk) [inline setup.py]: https://hynek.me/articles/conditional-python-dependencies/ has a way to do this for Python 3 only

**Comment by gordonje** [comment]
@jamesturk Got the conditional install/import of backports.csv working. I followed [Hynek Schlawack's advice](https://hynek.me/articles/conditional-python-dependencies/) pretty much to the letter, which meant adding some crufty stuff to `setup.py`. 

One thing to note, though: He claims that that the simple use of the environment markers in the `install_requires` arg to `setup()` will silently fail for setuptools 21 or greater, but I'm not seeing that. I'm running version 34 locally, and the use of the environment marker in `requirements_test.txt` worked. Also seems to be working on travis. 

TBH, this is much deeper into the weeds of setuptools than I have ever ventured, so I'm willing to trust an expert and believe that what I've just pushed will work for the widest range of users.

**Comment by jamesturk** [comment]
the conditional stuff looks alright to me, thanks!

I suppose the namespace stuff is alright to be in both places, @fgregg did you have a better solution for dedupe or was this what you did?

**Comment by fgregg** [comment]
@jamesturk this is pretty much what I did.

**Pull request merged**